hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogBlockMetadataScanner.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.log;
+
+import org.apache.hudi.avro.HoodieAvroReaderContext;
+import org.apache.hudi.common.engine.HoodieReaderContext;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.hudi.common.util.Option;
+
+import org.apache.avro.generic.IndexedRecord;
+
+import java.util.List;
+
+/**
+ * Scans a set of log files to extract metadata about the log blocks. It does not read the actual records.
+ */
+public class HoodieLogBlockMetadataScanner extends BaseHoodieLogRecordReader<IndexedRecord> {
+
+  public HoodieLogBlockMetadataScanner(HoodieTableMetaClient metaClient, List<String> logFilePaths, int bufferSize, Option<InstantRange> instantRange) {
+    super(getReaderContext(metaClient), metaClient, metaClient.getStorage(), logFilePaths, false, bufferSize, instantRange, false, false, Option.empty(), Option.empty(), true,
+        null, false);
+    scanInternal(Option.empty(), true);
+  }
+
+  private static HoodieReaderContext<IndexedRecord> getReaderContext(HoodieTableMetaClient metaClient) {
+    HoodieReaderContext<IndexedRecord> readerContext = new HoodieAvroReaderContext(metaClient.getStorage().getConf(), metaClient.getTableConfig(), Option.empty(), Option.empty());
+    readerContext.setHasLogFiles(true);
+    readerContext.setHasBootstrapBaseFile(false);
+    readerContext.setLatestCommitTime(metaClient.getActiveTimeline().lastInstant().map(HoodieInstant::requestedTime).orElseThrow(() -> new IllegalStateException("No completed instant found")));
```

Comment:
we should use the max instant time instead.

Reply:
What is the difference between this and lastInstant?

Reply:
the max instant is the upper threshold for log compaction I guess.

Reply:
Updated this to take in the maxInstant value from the caller to better max the current usage

Reply:
to give you practical use-case: 
in version 6 tables, when time travel query is issued, even though file slice might contain say 10 log files, we might be interested to read only up until instant time t5. since from the log file name, we can't decipher the delta commit time, this maxInstantTime will be useful. 

for snapshot reads also, the driver will determine the max instant time so that its consistent across all file slice reading instead of every file slice trying to infer on its own.